### PR TITLE
Support sending GuidanceResponse ETASU check to intermediary and keep…

### PR DESCRIPTION
**Pull request by jtquach1**
## Describe your changes

> Please include a summary of the changes and the related issue/task. Please also include relevant motivation and context. List any dependencies that are required for this change, including links to other pull requests/branches in other repositories if applicable.

* Added environment variables
* Support sending ETASU check to rems-intermediary and keep support for sending to rems-admin via environment variables
* Updated package-lock.json (used Node v21.7.3)

## Issue ticket number and Jira link

> Please include the Jira Ticket Number and Link for this issue/task.

https://jira.mitre.org/browse/REMS-712

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Ensure the target / base branch for any feature PR is set to `dev` not main (the only exception to this is releases from `dev` and hotfix branches)

## Checklist for conducting a review
- [ ] Review the code changes and make sure they all make sense and are necessary. 
- [ ] Pull the PR branch locally and test by running through workflow and making sure everything works as it is supposed to. 

## Workflow 

Owner of the Pull Request will be responsible for merge after all requirements are met, including approval from at least one reviewer. Additional changes made after a review will dismiss any approvals and require re-review of the additional updates. Auto merging can be enabled below if additional changes are likely not to be needed. The bot will auto assign reviewers to your Pull Request for you.

### CODE REVIEW

File: backend/src/routes/doctorOrders.js [hunk 1]
```diff
@@ -359,7 +358,7 @@ async function parseNCPDPScript(newRx) {
     dispenseStatus: 'Pending'
   };
 
-  const isRemsDrug = !!getRemsAdminFhirUrl(incompleteOrder);
+  const isRemsDrug = !!getRemsAdminEtasuUrl(incompleteOrder);
```
Comment (plarocque4) [inline backend/src/routes/doctorOrders.js:361]: How would this work if there was an intermediary setup and not every drug with a REMS listed in the configuration? Do we need to find a way to query the intermediary to check if it is a REMS drug?

File: backend/env.json [hunk 2]
```diff
@@ -53,5 +53,13 @@
...
     "type": "string",
     "default": "http://localhost:8080/test-ehr/script/rxfill"
+  },
+  "USE_INTERMEDIARY": {
+    "type": "boolean",
+    "default": true
+  },
+  "INTERMEDIARY_ETASU_MET": {
+    "type": "string",
+    "default": "http://localhost:3003/4_0_0/GuidanceResponse/$rems-etasu"
```
Comment (plarocque4) [inline backend/env.json:63]: Why not just make this INTERMEDIARY_FHIR_URL and then add on the /GuidanceResponse/$rems-etasu this this as well as what is read from the table in doctorOrders.js.

File: backend/src/routes/doctorOrders.js [hunk 3]
```diff
@@ -225,18 +225,20 @@ router.delete('/api/deleteAll', async (req, res) => {
...
   });
-  return remsDrug?.remsAdminFhirUrl;
+  const baseUrl = remsDrug?.remsAdminFhirUrl;
+  const remsAdminFhirUrl = baseUrl + '/GuidanceResponse/$rems-etasu';
+  return baseUrl ? remsAdminFhirUrl : null;
 };
 
 const getGuidanceResponse = async order => {
-  const remsAdminFhirUrl = getRemsAdminFhirUrl(order);
+  const etasuUrl = env.USE_INTERMEDIARY ? env.INTERMEDIARY_ETASU_MET : getRemsAdminEtasuUrl(order);
```
Comment (plarocque4) [inline backend/src/routes/doctorOrders.js:239]: I think you should move this logic into the getRemsAdminUrl function above and add on the /GuidanceResponse/$rems-etasu in that function to every FHIR URL including what is set for the intermediary.

File: backend/src/routes/doctorOrders.js (hunk 1 above)
Comment (jtquach1) [reply backend/src/routes/doctorOrders.js:361]: That's a good question- I suppose the pharmacy wouldn't know innately if Tylenol/ibuprofen was a REMS drug. Maybe if the intermediary had an endpoint that returns a boolean based on what the DoctorOrder is (drug id, is using the rxnorm system url), if the drug isn't a REMS drug then we use whatever default FHIR url the pharmacy has coded in.

File: backend/env.json (hunk 2 above)
Comment (jtquach1) [reply backend/env.json:63]: I suppose I could refactor the getRemsAdminEtasuUrl function so it finds the ETASU url if the appropriate environment variable is set. Is `GuidanceResponse/$rems-etasu` a fixed string from the REMS IG or does it follow a typical FHIR naming standard for endpoints? Or is that just specific to the `@projecttacoma/node-fhir-server-core` package?

File: backend/src/routes/doctorOrders.js (hunk 1 above)
Comment (jtquach1) [reply backend/src/routes/doctorOrders.js]: is your question/feature suggestion is related to https://jira.mitre.org/browse/REMS-709?

**Comment by plarocque4** [comment]
This is much improved and works well, just address the last comment from above please.

**Comment by jtquach1** [comment]
> This is much improved and works well, just address the last comment from above please.

Done!

**Comment by plarocque4** [comment]
Acetaminophen has no REMS but I am unable to verify the order, must be something wrong with ETASU checking on a drug with no ETASU/REMS?

**Comment by jtquach1** [comment]
> Acetaminophen has no REMS but I am unable to verify the order, must be something wrong with ETASU checking on a drug with no ETASU/REMS?

Fixed in https://github.com/mcode/pims/pull/86/commits/fd2bc8001d3f2a5890e3c948a0f81b4602c91231!

**Review by plarocque4 — APPROVED** [review]
looks good

**Pull request merged**